src/sentry/workflow_engine/processors/data_condition_group.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any
+
+from sentry.utils.function_cache import cache_func_for_models
+from sentry.workflow_engine.models import DataCondition, DataConditionGroup
+from sentry.workflow_engine.types import ProcessedDataConditionResult
+
+
+@cache_func_for_models(
+    [(DataConditionGroup, lambda group: (group.id,))],
+)
+def get_data_condition_group(data_condition_group_id: int) -> DataConditionGroup | None:
+    try:
+        group = DataConditionGroup.objects.get(id=data_condition_group_id)
+    except DataConditionGroup.DoesNotExist:
+        group = None
+    return group
```

Comment:
Since this is just fetching a `DataConditionGroup` by its id, you might be better off using https://github.com/getsentry/sentry/blob/16eee6355ae0905ee57913773a61cb02800382c4/src/sentry/models/group.py#L582